controllers/managedocs_controller.go

```diff
@@ -2045,3 +2096,31 @@ func (r *ManagedOCSReconciler) setDeviceSetCount(deviceSet *ocsv1.StorageDeviceS
 		deviceSet.Count = currDeviceSetCount
 	}
 }
+
+func (r *ManagedOCSReconciler) reconcileODFSubscription() error {
+	subList := &opv1a1.SubscriptionList{}
+	if err := r.list(subList); err != nil {
+		return fmt.Errorf("Unable to list subscriptions %v", err)
+	}
+	odfSub := &opv1a1.Subscription{}
+	for index := range subList.Items {
+		item := &subList.Items[index]
+		if item.Spec.Package == odfOCMBundleName {
+			odfSub = item
+			break
+		}
+	}
+	if odfSub == nil {
+		return fmt.Errorf("Failed to find package name %s", odfOCMBundleName)
+	}
+	if odfSub.Spec.Channel == desiredODFSubscriptionChannel {
+		return nil
+	}
+	r.Log.Info("Initiating ODF minor version upgrade", "current release", odfSub.Spec.Channel, "target release", desiredODFSubscriptionChannel)
```

Comment:
I think we should be more specific with the information line in the log
```suggestion
	r.Log.Info("Initiating ODF minor version upgrade by switching channels", "current channel", odfSub.Spec.Channel, "new channel", desiredODFSubscriptionChannel)
```

Reply:
Updated